### PR TITLE
Add support for Razer Naga V2 HyperSpeed (Receiver)

### LINKED ISSUE #1966: Support for Razer Naga V2 Hyperspeed
### Device Info

- Name: Razer Naga V2 Hyperspeed
- Product Number: RZ01-03600100
- Model Number: RZ01-0360

### Device Information
```Bus 001 Device 007: ID 1532:00b4 Razer USA, Ltd Razer Naga V2 HyperSpeed
Device Descriptor:
  bLength                18
  bDescriptorType         1
  bcdUSB               2.00
  bDeviceClass            0 
  bDeviceSubClass         0 
  bDeviceProtocol         0 
  bMaxPacketSize0        64
  idVendor           0x1532 Razer USA, Ltd
  idProduct          0x00b4 
  bcdDevice            1.00
  iManufacturer           1 Razer
  iProduct                2 Razer Naga V2 HyperSpeed
  iSerial                 3 000000000000
  bNumConfigurations      1
  Configuration Descriptor:
    bLength                 9
    bDescriptorType         2
    wTotalLength       0x0054
    bNumInterfaces          3
    bConfigurationValue     1
    iConfiguration          4 Default configuration
    bmAttributes         0xa0
      (Bus Powered)
      Remote Wakeup
    MaxPower              100mA
    Interface Descriptor:
      bLength                 9
      bDescriptorType         4
      bInterfaceNumber        0
      bAlternateSetting       0
      bNumEndpoints           1
      bInterfaceClass         3 Human Interface Device
      bInterfaceSubClass      1 Boot Interface Subclass
      bInterfaceProtocol      2 Mouse
      iInterface              0 
        HID Device Descriptor:
          bLength                 9
          bDescriptorType        33
          bcdHID               1.11
          bCountryCode            0 Not supported
          bNumDescriptors         1
          bDescriptorType        34 Report
          wDescriptorLength      97
         Report Descriptors: 
           ** UNAVAILABLE **
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x81  EP 1 IN
        bmAttributes            3
          Transfer Type            Interrupt
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0040  1x 64 bytes
        bInterval               1
    Interface Descriptor:
      bLength                 9
      bDescriptorType         4
      bInterfaceNumber        1
      bAlternateSetting       0
      bNumEndpoints           1
      bInterfaceClass         3 Human Interface Device
      bInterfaceSubClass      1 Boot Interface Subclass
      bInterfaceProtocol      1 Keyboard
      iInterface              0 
        HID Device Descriptor:
          bLength                 9
          bDescriptorType        33
          bcdHID               1.11
          bCountryCode            0 Not supported
          bNumDescriptors         1
          bDescriptorType        34 Report
          wDescriptorLength     314
         Report Descriptors: 
           ** UNAVAILABLE **
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x82  EP 2 IN
        bmAttributes            3
          Transfer Type            Interrupt
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0040  1x 64 bytes
        bInterval               1
    Interface Descriptor:
      bLength                 9
      bDescriptorType         4
      bInterfaceNumber        2
      bAlternateSetting       0
      bNumEndpoints           1
      bInterfaceClass         3 Human Interface Device
      bInterfaceSubClass      1 Boot Interface Subclass
      bInterfaceProtocol      1 Keyboard
      iInterface              0 
        HID Device Descriptor:
          bLength                 9
          bDescriptorType        33
          bcdHID               1.11
          bCountryCode            0 Not supported
          bNumDescriptors         1
          bDescriptorType        34 Report
          wDescriptorLength      61
         Report Descriptors: 
           ** UNAVAILABLE **
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x83  EP 3 IN
        bmAttributes            3
          Transfer Type            Interrupt
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0040  1x 64 bytes
        bInterval               1
Device Status:     0x0000
  (Bus Powered)
```

```total 0
drwxr-xr-x 2 root root 300 Nov 29 17:54 .
drwxr-xr-x 4 root root 800 Nov 29 19:13 ..
lrwxrwxrwx 1 root root  10 Nov 29 17:54 usb-046d_Logitech_StreamCam_F0023F65-event-if00 -> ../event13
lrwxrwxrwx 1 root root   9 Nov 29 17:54 usb-04d9_USB-HID_Keyboard-event-if02 -> ../event5
lrwxrwxrwx 1 root root   9 Nov 29 17:54 usb-04d9_USB-HID_Keyboard-event-kbd -> ../event2
lrwxrwxrwx 1 root root   9 Nov 29 17:54 usb-04d9_USB-HID_Keyboard-if02-event-kbd -> ../event6
lrwxrwxrwx 1 root root  10 Nov 29 17:54 usb-Razer_Razer_Naga_V2_HyperSpeed_000000000000-event-if01 -> ../event10
lrwxrwxrwx 1 root root   9 Nov 29 17:54 usb-Razer_Razer_Naga_V2_HyperSpeed_000000000000-event-mouse -> ../event7
lrwxrwxrwx 1 root root   9 Nov 29 17:54 usb-Razer_Razer_Naga_V2_HyperSpeed_000000000000-if01-event-kbd -> ../event8
lrwxrwxrwx 1 root root   9 Nov 29 17:54 usb-Razer_Razer_Naga_V2_HyperSpeed_000000000000-if01-event-mouse -> ../event9
lrwxrwxrwx 1 root root   9 Nov 29 17:54 usb-Razer_Razer_Naga_V2_HyperSpeed_000000000000-if01-mouse -> ../mouse2
lrwxrwxrwx 1 root root  10 Nov 29 17:54 usb-Razer_Razer_Naga_V2_HyperSpeed_000000000000-if02-event-kbd -> ../event11
lrwxrwxrwx 1 root root   9 Nov 29 17:54 usb-Razer_Razer_Naga_V2_HyperSpeed_000000000000-mouse -> ../mouse1
lrwxrwxrwx 1 root root   9 Nov 29 17:54 usb-Valve_Software_Steam_Controller-event-mouse -> ../event3
lrwxrwxrwx 1 root root   9 Nov 29 17:54 usb-Valve_Software_Steam_Controller-mouse -> ../mouse0
```

### PR DESCRIPTION
**Pull request by xonvanetta**
Co-authored-by: rhaamo <dashie@sigpipe.me>

Resolves: #1966

Copied alot from PR #2035 and fixed that dpi was set using openrazer python api

Is there anything else I should test/verify?

### CODE REVIEW

**Comment by xonvanetta** [comment]
I see formating issues, will fix

File: driver/razermouse_driver.h [hunk 1]
```diff
@@ -90,6 +90,7 @@
 #define USB_DEVICE_ID_RAZER_HYPERPOLLING_WIRELESS_DONGLE 0x00B3
 #define USB_DEVICE_ID_RAZER_DEATHADDER_V3_PRO_WIRED 0x00B6
 #define USB_DEVICE_ID_RAZER_DEATHADDER_V3_PRO_WIRELESS 0x00B7
+#define USB_DEVICE_ID_RAZER_NAGA_V2_HYPERSPEED_RECEIVER 0x00B4
```
Comment (z3ntu) [inline driver/razermouse_driver.h]: move this up just below `USB_DEVICE_ID_RAZER_HYPERPOLLING_WIRELESS_DONGLE` so the list is sorted by value

**Comment by z3ntu** [comment]
Generally looks good! 

> Is there anything else I should test/verify?

So everything shown in e.g. polychromatic works correctly?

**Comment by xonvanetta** [comment]
>So everything shown in e.g. polychromatic works correctly?

I did a really small test but I wanna compare it more when I get a little more time

One thing that did not work or rather I don't know how it works is this low battery mode, didn't work on Windows either when I tried

**Comment by z3ntu** [comment]
> I did a really small test but I wanna compare it more when I get a little more time

Sure, that's fine. Just so you know I'm planning to make a release maybe in the upcoming weekend (or the one after) so would be nice if this PR could be included then.

**Comment by xonvanetta** [comment]
Fixed conflict and comment from PR

>Sure, that's fine. Just so you know I'm planning to make a release maybe in the upcoming weekend (or the one after) so would be nice if this PR could be included then.

Sounds good, I will try make a window tomorrow to try test it out with polychromatic more and verify that it works as intended

**Comment by xonvanetta** [comment]
Testing this PR:

```
$ polychromatic-cli --version
Polychromatic 0.8.2
Python: 3.11.5
```

- [x] DPI
    - [x] get current dpi
    - [x] set only x or y
    - [x] set stages 
    - [x] get stages
- [x] Sleep mode
- [ ] Low power mode, how does this even work? 
- [x] poll rate

Did some tests and everything works except `low power mode`, tested with windows and also no difference. default stages in polychromatic are [preset on max dpi or calculated if not found](https://github.com/polychromatic/polychromatic/blob/e983d5a0f0e8ab2c44250cd11cb98eee42020bf1/polychromatic/backends/openrazer.py#L346-L350). Get stages work great using the python API

**Comment by z3ntu** [comment]
Sounds sane. Also confirmed on the product page that there's no RGB on this mouse. I'll do a final review soon :)

**Review by z3ntu — APPROVED** [review]
LGTM, thanks!

**Pull request merged**